### DISCUSSION TITLE
Idiomatic "default route" at index route Outlet

### DISCUSSION THREAD

**Discussion by adamhp** [remix-run/remix#1907] [Q&A] [answered]
Say you have the main `routes/index.tsx` route, and you put all of your layout there, with an `<Outlet/>`. Is there any idiomatic or straightforward way of making sure there is a default rendering for the Outlet? For example, if I wanted the user to see their dashboard when the path is `/`, I can't exactly do that. I'm essentially using `index.tsx` to have a sidebar with a main content window that will change on route change, but I want there to be at least _something_ when the user is just at the `/` route, rather than having to reroute them to `/home` or `/dashboard` or something. 

Any ideas?

EDIT: Should have searched _FIRST_. Appears this may be a duplicate of #1490. I'll give that a shot and report back!

**Comment by sergiodxa** [comment, marked as answer]
I think if you only want a layout but not add segments to the URL (I suppose that's why you used `routes/index` then you can use a pathless layout route.

Create a file `routes/__layout.tsx` and a folder `routes/__layout`, inside the file put your sidebar and the Outlet, inside `routes/__layout` folder put any route using that layout, if you have a `routes/__layout/index.tsx` it will render at `/` but using the layout, if you have `routes/__layouts/dashboard.tsx` it will render at `/dashboard` and inside the layout.

See more on https://remix.run/docs/en/v1/api/conventions#pathless-layout-routes

**Reply by adamhp** [reply]
This is very helpful! I had actually read those docs but didn't quite grok the first time. I think your explanation made it click. Thanks!

**Reply by adamhp** [reply]
Works perfectly! And has the nice side effect that I can render my login and other pages that don't need the layout separately. I also ended up using the splat route `$.tsx` for the dashboard, so that is just the default catch-all render 👌🏻 ... Thanks again!

**Comment by arnaudambro** [comment]
In have another use case : I'm trying to create a social network, where `/` would be the route for the feed to be shown, and `/new-post` to be the route to create a new post, **_above_** the feed, like when you create a new post on LinkedIn.
Then all the other routes (`/profile`, `/onboarding`...) are not on top of the feed.

I can't find the proper route folder setup to do so.
I tried to follow the pattern in #1474 but for now it's not successful...

I'm trying
```
app/
├── routes/
│   ├── __home/
│   │   ├── $.jsx
│   │   └── new-post.jsx
│   ├── __home.jsx
└── root.jsx
```

but nothing is rendered

Then I try to replace the splat route `__home/index.jsx
```
app/
├── routes/
│   ├── __home/
│   │   ├── index.jsx
│   │   └── new-post.jsx
│   ├── __home.jsx
└── root.jsx
```
But the `/new-post` route doesn't render on top of the feeds anymore

I don't know where next to go !

**Reply by arnaudambro** [reply]
linked to #1490

**Reply by silvenon** [reply]
Sounds like the feed should be in the layout route, i.e. in `__home.jsx` instead of `__home/index.jsx`. And if there's nothing you want to add in `__home/index.jsx` over what `__home.jsx` already renders, you could export a component that returns `null`.

**Reply by arnaudambro** [reply]
but then I have an action there, for likes/unlikes, and I have 
`Actions are not supported in layout routes: routes/__home`

It means I need/have to to create specific routes for those, no other choice  ?

**Reply by silvenon** [reply]
And if you have trouble with loaders, put the same logic in loaders of `__home/index.jsx` and `__home/new-post.jsx`, but access that data from `__home.jsx` to create the feed, using the `useMatches` hook.

**Reply by silvenon** [reply]
> but then I have an action there, for likes/unlikes, and I have
> `Actions are not supported in layout routes: routes/__home`

I think the only way is for `__home/new-post.jsx` to export that same `action`. You can import it from `__home/index.jsx`.

This is a very interesting use case — you want one route to be built upon another one. If I was you I would open a new discussion just in case to see what people have to say.

**Reply by silvenon** [reply]
Actually no, what am I saying. Likes/unlikes can just always send requests to `__home/index.jsx` (if that's where you put your `action`) regardless of the current route, you just need to add the `action` prop to your form: `<Form method="post" action="/">`

Alternatively you can create a separate resource route that would function as an API. One can be for likes, another one for something else etc. It's up to you how you want to organize this logic.

Makes sense?

**Reply by arnaudambro** [reply]
I just added a `/post-actions` as a ressource route, and it's working good now.
But what I like in remix is to have actions near the component, and it's not the best !

Let me try with your last solution, I'll write here if it looks good

**Reply by arnaudambro** [reply]
OK, your solution seems to be the best so far that I explored, thanks !

So, posts loader is in `/routes/__home.jsx`, and post actions (like/unlike/delete/comment/whatever) are in `/routes/__home/index.jsx`, which file is exporting `null` as default function.

Yet, maybe I'll end-up create a separate resource route called `/routes/post-actions.jsx`, it seems more logical to see them here than in `/routes/__home/index.jsx`... but that's another thing, that's up to me how to organize my logic as you say.

Fact is
-> we "solved" the problem with this `/routes/__home/index.jsx`
-> I still miss the ability to put `action` closed to my `Posts`..

Thanks @silvenon !

**Reply by silvenon** [reply]
Yeah, I get what you mean, I hope you land on a structure you're happy with. Not sure what your exact situation is, but you can export loading logic from your component files. Assuming Posts is a regular component, you can export a function like `likePost` next to the component, and use it in an action.

Good luck!
